pkg/endpoint/bpf.go

```diff
@@ -161,7 +161,7 @@ func (e *Endpoint) writeHeaderfile(prefix string, owner Owner) error {
 	fw.WriteString(common.FmtDefineAddress("LXC_MAC", e.LXCMAC))
 	fw.WriteString(common.FmtDefineAddress("LXC_IP", e.IPv6))
 	if e.IPv4 != nil {
-		fmt.Fprintf(fw, "#define LXC_IPV4 %#x\n", binary.BigEndian.Uint32(e.IPv4))
+		fmt.Fprintf(fw, "#define LXC_IPV4 %#x\n", byteorder.HostToNetworkSliceU32(e.IPv4))
```

Comment:
The BPF program in master expects `LXC_IPV4` to be in host byte order at the moment. Let's keep it as-is for this PR and we can optimize this later. The test is failing because you would also have to change the bpf side first.